### PR TITLE
Tests for test connection in datasource wizard

### PR DESCRIPTION
**Pull request by honza-kasik**
* Ready for 2.9.0.Final
* Rewritten old tests

### CODE REVIEW

**Comment by honza-kasik** [comment]
@zebrik @rhatlapa Can you please take a look at this PR? The RFE is in 'Verification TODO' state.

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/AbstractTestConnectionTestCase.java [hunk 1]
```diff
@@ -9,81 +9,194 @@
...
 
 /**
  * @author jcechace
  */
 public abstract class AbstractTestConnectionTestCase {
 
+    /**
+     * Hook which method will be executed after test connection was performed in wizard
+     */
+    protected interface Hook {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/AbstractTestConnectionTestCase.java]: More descriptive name would be better :-)

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/AbstractTestConnectionTestCase.java [hunk 2]
```diff
@@ -9,81 +9,194 @@
...
 
-    protected void testConnection(String name, boolean expected) {
-        datasourcesPage.view(name);
-        Console.withBrowser(browser).waitUntilLoaded();
-
+    /**
+     * Tests connection in datasource view
+     * @param expected true if connection is expected to succeed, false otherwise
+     */
+    protected void testConnection(boolean expected) {
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/AbstractTestConnectionTestCase.java]: Better descriptive name, to make clear in what context the connection is tested

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/DomainTestConnectionTestCase.java [hunk 3]
```diff
@@ -1,112 +1,82 @@
...
-    }
 
     @AfterClass
-    public static void tearDown() throws CommandFailedException { // remove datasources when finished
+    public static void tearDown() throws InterruptedException, TimeoutException, IOException {
         try {
-            dsOps.removeDataSource(dsNameValid);
-            new DataSourcesOperations(fullHaClient, "full-ha").removeDataSource(dsSameNameValid); ///
-            dsOps.removeDataSource(dsSameNameInvalid);
+            administration.reloadIfRequired();
```
Comment (rhatlapa) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/datasources/DomainTestConnectionTestCase.java]: ??? teardown method which calls only reload seems as hack for something which should be already handled, but OK.

File: common/src/main/java/org/jboss/hal/testsuite/fragment/WindowFragment.java [hunk 4]
```diff
@@ -87,13 +89,17 @@ public WindowState clickSave() {
...
     public void cancel() {
-        String label = PropUtils.get("modals.window.cancel.label");
-        clickButton(label);
+        clickCancel();
 
         waitUntilClosed();
         closed = true;
     }
 
+    public void clickCancel() {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/WindowFragment.java]: Doesn't seem as something which should be public, I would say, it should be rather private.
If it is required by cancelAndDismissReloadRequiredWindow, it should be most likely improved in order for this method not being required public

**Review by rhatlapa — COMMENTED** [review]
Generally LGTM, just suggestions for rather minor improvements to the API.

**Comment by rhatlapa** [comment]
Thanks, LGTM, merging.

**Pull request merged**